### PR TITLE
Rename PseudoCode widget to Decompiler

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
**Detailed description**

Rename the widget from "Pseudocode" to "Decompiler" 

image

### CODE REVIEW

**Comment by karliss** [comment]
I haven't tested it yet, but doesn't this require settings upgrade to prevent breaking layout due to name change?

**Comment by ITAYC0HEN** [comment]
I think it does, but isn't it the price we need to pay for it?

**Comment by ITAYC0HEN** [comment]
so, what do you think of the settings update? Any way to overcome this or c'est la vie?

**Comment by karliss** [comment]
Don't we have settings upgrade mechanism already implemented? You shouldnt need to do more than add a function performing x to. X+1 upgrade step and increase the version by one. Rest of work should be handled by existing code.

**Comment by karliss** [comment]
See https://github.com/radareorg/cutter/blob/06aceaf9b10ff466d9bf85ff12c0660e64e39da7/src/Main.cpp#L42

**Comment by ITAYC0HEN** [comment]
@karliss @xarkes please check now after the changes :)

**Comment by karliss** [comment]
There is probably a reference to dock names in binary window layout data which makes the upgrade step less useful than I thought, sorry about that. There is nothing we can easily and cleanly do about it.

File: src/dialogs/preferences/AsmOptionsWidget.cpp [hunk 1]
```diff
@@ -250,7 +250,7 @@ void AsmOptionsWidget::asmComboBoxChanged(int index)
     // Check if ESIL enabled
     Config()->setConfig("asm.esil", index == 1);
 
-    // Check if Pseudocode enabled
+    // Check if Decompiler enabled
```
Comment (thestr4ng3r) [inline src/dialogs/preferences/AsmOptionsWidget.cpp]: This makes no sense for asm.pseudo, but the comment is redundant anyway

**Comment by thestr4ng3r** [comment]
> There is probably a reference to dock names in binary window layout data which makes the upgrade step less useful than I thought, sorry about that. There is nothing we can easily and cleanly do about it.

Yes exactly, and this will not be possible to be resolved until implementing the json format proposed by you, so lgtm for now.

**Comment by karliss** [comment]
I wasn't planning to change the way dock layout gets serialized. That's a fairly complex code and it would be good to reuse functionality provided by dock and window management library. It's something to pay attention when evaluating alternatives for qt docking widgets.

**Pull request merged**